src/provider/provider_file_memory.c

```diff
@@ -545,6 +553,14 @@ static umf_result_t file_get_ipc_handle_size(void *provider, size_t *size) {
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
+    file_memory_provider_t *file_provider = (file_memory_provider_t *)provider;
+    if (!file_provider->IPC_enabled) {
```

Comment:
I am not sure that the `file_get_ipc_handle_size()` function should fail if the `IPC_enabled` flag is `false`.
I am thinking about the scenario when on the consumer side, the user creates a file provider without required flags (`UMF_MEM_MAP_SHARED` and `UMF_MEM_MAP_SYNC`) and just wants to use this provider to open IPC handles and wants to know expected IPC handle size (MPI uses our API in that way).

Reply:
More general question: do we want to allow to open IPC handles by the file provider instance that does not have the abovementioned flags set?

Reply:
The consumer also has to have the correct visibility mode set, otherwise it does not work:
https://github.com/ldorau/unified-memory-framework/commit/7cd29a75c0dcb6f4e139a5250a1e0d28c5a28924
https://github.com/ldorau/unified-memory-framework/actions/runs/11056477461/job/30718352219

Reply:
> I am not sure that the `file_get_ipc_handle_size()` function should fail if the `IPC_enabled` flag is `false`.
> I am thinking about the scenario when on the consumer side, the user creates a file provider without required flags (`UMF_MEM_MAP_SHARED` and `UMF_MEM_MAP_SYNC`) and just wants to use this provider to open IPC handles and wants to know expected IPC handle size (MPI uses our API in that way).

When the consumer opens the IPC handle with the MAP_PRIVATE flag, it will not work as you can see in the example above.

Reply:
> When the consumer opens the IPC handle with the `MAP_PRIVATE` flag, it will not work:
> [ldorau@7cd29a7](https://github.com/ldorau/unified-memory-framework/commit/7cd29a75c0dcb6f4e139a5250a1e0d28c5a28924)
> https://github.com/ldorau/unified-memory-framework/actions/runs/11056477461/job/30718352219

That is exactly my question: do we want such behavior? Why cannot we open the handle in such a case?

Reply:
This build was run on the current main branch, without this patch.
We do open the handle and it does not work, because it was mapped with `MAP_PRIVATE`.

Reply:
```
[consumer] Received the IPC handle from the producer (4112 bytes)
[PID:[165](https://github.com/ldorau/unified-memory-framework/actions/runs/11056477461/job/30718352219#step:12:166)72  TID:16572  DEBUG UMF] utils_mmap_file: file mapped with the MAP_PRIVATE flag
[consumer] Opened the IPC handle received from the producer
[consumer] Read the number from the producer's shared memory: 140722523604400
[consumer] Wrote a new number directly to the producer's shared memory: 70361261802200
[consumer] Sent a response message to the producer
[producer] Received the consumer's response: "This is the consumer. I just wrote a new number directly into your shared memory!"
[producer] ERROR: The consumer did NOT write the correct value (the old one / 2 = 70361261802200) to my shared memory: 140722523604400
[consumer] Closed the IPC handle received from the producer
[producer] Put the IPC handle
[producer] Shutting down (status ERROR) ...
[consumer] Shutting down (status OK) ...
```

Reply:
So the scenario, you want to use, does not work. The consumer also has to map with `MAP_SHARED`.

Reply:
I think I got the point: the `UMF_MEM_MAP_SHARED` and `UMF_MEM_MAP_SYNC` flags are used not only in the provider's alloc function implementation but also in the "open IPC Handle" implementation. In both cases, mmap should be called with these flags.